Author: Daniel Baumann <daniel.baumann@progress-technologies.net>
Description:
 Only create rootfs directory where required. In general, lxc-create
 should not mess with this and leave it entirely up to the templates
 to create it.

diff -Naurp lxc.orig/src/lxc/lxc-create.in lxc/src/lxc/lxc-create.in
--- lxc.orig/src/lxc/lxc-create.in	2012-03-15 16:58:33.069832600 +0100
+++ lxc/src/lxc/lxc-create.in	2012-04-10 15:31:32.575773434 +0200
@@ -255,8 +255,8 @@ else
 fi
 
 # Create the fs as needed
-[ -d "$rootfs" ] || mkdir $rootfs
 if [ $backingstore = "lvm" ]; then
+    [ -d "$rootfs" ] || mkdir $rootfs
     lvcreate -L $fssize -n $lvname $vgname || exit 1
     udevadm settle
     mkfs -t $fstype $rootdev || exit 1
